### PR TITLE
Fixes #10185: Modified the srepr docstring

### LINKED ISSUE #10185: Clarification about srepr in the docs
Quoting from Gitter chat

> srepr gives the “repr” form, which is what reprwould normally give
but for SymPy we don’t actually use it for `__repr__` because it’s way too verbose
and on top of that when you print a list or a tuple it calls repr on the elements instead of str, so a list of sympy expressions would always have the repr output
that page should probably mention all of this. it could be a lot clearer
oh I guess that’s the tutorial. For the tutorial I would just replace the “repr” heading with “srepr”. But in the docs for srepr it should have the discussion about the repr builtin

We should rename the `repr` section in the tutorial to `srepr`, and add some in-depth discussion (basically what I wrote above) to the `srepr` docstring, which appears [here](http://docs.sympy.org/dev/modules/printing.html?highlight=srepr#sympy.printing.repr.srepr).

### PR DESCRIPTION
**Pull request by sampadsaha5**
Fixes #10185.
Ping @asmeurer

### CODE REVIEW

**Comment by mafiya69** [comment]
duplicate of #10215

**Comment by sampadsaha5** [comment]
@asmeurer Is anything else needed to add in the srepr docstring?

File: doc/src/modules/printing.rst [hunk 1]
```diff
@@ -307,14 +307,29 @@ This class implements Python printing. Usage::
...
+
+Example::
+
+  >>> repr(5*x**3 + sin(x))
+  '5*x**3 + sin(x)'
+
+  >>> srepr(5*x**3 + sin(x))
+  "Add(Mul(Integer(5), Pow(Symbol('x'), Integer(3))), sin(Symbol('x')))"
+
+``Srepr()`` gives the ``repr`` form, which is what reprwould normally give
```
Comment (asmeurer) [inline doc/src/modules/printing.rst]: `srepr` should be all lowercase.

Comment (asmeurer) [inline doc/src/modules/printing.rst]: You are missing a space ("repr would")

File: doc/src/modules/printing.rst [hunk 2]
```diff
@@ -307,14 +307,29 @@ This class implements Python printing. Usage::
...
+but for SymPy we don’t actually use ``Srepr()`` for ``__repr__`` because it’s 
+is so verbose, it is unlikely that anyone would want it called by default.
```
Comment (asmeurer) [inline doc/src/modules/printing.rst]: The other reason is that lists call repr on their elements, like `print([a, b, c])` calls `repr(a)`, `repr(b)`, `repr(c)`. So if we used `srepr` for `__repr__` any list with SymPy objects would include the srepr form, even if you used `str` or `print`.

**Comment by sampadsaha5** [comment]
Thanks you @asmeurer for the suggestions. I have changed the docstring according to your suggestions.

**Comment by sampadsaha5** [comment]
Thank you @certik

**Pull request merged**